File: leetcode/recursion/powerOfThree.cpp
/**
 * @file powerOfThree.cpp
 * @author Rahul Awasthi (awasthir@pm.me)
 * 
 * @brief 
 * try out the problem here:
 * https://leetcode.com/problems/power-of-three/
 * 
 */

class Solution {
    bool iterative(int n) {
        if (n < 1) return false;
        while (n != 1) {
            if (n % 3 != 0) return false;
            n /= 3;
        }
        return true;
    }
public:
    bool isPowerOfThree(int n) {
        return iterative(n);
    }
};

File: test-rep/powerOfThree.py
"""

    @file powerOfThree.cpp
    @author Rahul Awasthi (awasthir@pm.me)

    @brief 
    try out the problem here:
    https://leetcode.com/problems/power-of-three

"""

def power_of_three(n):
    if (n < 1):return False
    while (n !=1):
        if (n % 3 == 1):
            return False
        n /= 3
    return True

x = [3**i for i in range(0, 12)]
print(x)
for i in x:
    print(power_of_three(i))